device/src/peripherals.rs

```diff
@@ -256,11 +256,10 @@ impl<'a> DevicePeripherals<'a> {
         .with_sda(&mut peripherals.GPIO4)
         .with_scl(&mut peripherals.GPIO5);
 
-        let mut capsense = CST816S::new_esp32(i2c, &mut peripherals.GPIO2, &mut peripherals.GPIO3);
-        capsense.setup(&mut delay).unwrap();
+        let capsense = CST816S::new_esp32(i2c, &mut peripherals.GPIO2, &mut peripherals.GPIO3);
 
-        // Register the capsense instance with the interrupt handler
-        let touch_receiver = frostsnap_cst816s::interrupt::register(capsense, &mut io);
+        // Register the capsense instance with the interrupt handler (includes setup)
```

Comment:
Why are we changing where the setup is done? What if we change the code not to use the interrupt handler in some circumstances? Then it won't run. The register function should only be for interrupt handling.